### PR TITLE
feature/CLAN6-291-fallback-missing-widget

### PR DESCRIPTION
**Pull request by jplucinski**
- missing types are replaced with "WhiteSpaceWidget"

### CODE REVIEW

File: cogboard-webapp/src/components/widgets/index.js [hunk 1]
```diff
@@ -177,3 +177,9 @@ const widgetTypes = {
 };
 
 export default widgetTypes;
+
+export function getWidgetType(widgetType) {
+  return widgetTypes.hasOwnProperty(widgetType)
+    ? widgetTypes[widgetType]
+    : widgetTypes['WhiteSpaceWidget'];
+}
```
Comment (Southeaven) [inline cogboard-webapp/src/components/widgets/index.js]: Please move this function outside `/widgets/index.js` (for example `helpers.js`). This file is used only for registering new widgets.

File: cogboard-webapp/src/components/Widget/index.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ const Widget = ({ id, index }) => {
     ...widgetTypeData
   } = widgetData;
   const { expandContent } = widgetTypeData;
-  const widgetTypeConfig = widgetTypes[type];
+  const widgetTypeConfig = getWidgetType(type);
```
Comment (szymon-owczarzak) [inline cogboard-webapp/src/components/Widget/index.js]: `const widgetTypeConfig = widgetTypes[type] | widgetTypes['WhiteSpaceWidget'];`

Comment (jplucinski) [reply cogboard-webapp/src/components/Widget/index.js]: True, I though about this. In my opinion, introducing helper function will prevent developers from using `widgetTypes[type]` explicitly.

File: cogboard-webapp/src/components/widgets/index.js (hunk 1 above)
Comment (jplucinski) [reply cogboard-webapp/src/components/widgets/index.js]: Fixed

File: cogboard-webapp/src/components/Widget/index.js (hunk 2 above)
Comment (jplucinski) [reply cogboard-webapp/src/components/Widget/index.js]: Done.

**Pull request merged**